wireapp/wire-ios-mono
# fix: Don't try to share non-existing logs - WPB-9948

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9948" title="WPB-9948" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-9948</a>  Sending logs - when email client is not configured
  </td></table>
  <br />
 

















### Issue

Currently when sending logs via a share sheet we attempt to attach all logging URLs derived from `LogFileDestination` and `ZMSLog`. However in some cases no logs have actually been created, which unsurprisingly can cause sending logs to fail.

This PR:

- Updates `DebugLogSender.debugLogs` to filter out urls for files that do not exist. It also renames it to `existingDebugLogs` to emphasize this and makes it a function for testing purposes.
- Makes `cachesDirectory` non optional which simplifies a bunch of dependent code as previously discussed. My assumption here is that `FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask)[0]` will always succeed. This is based on my experience and Apples implementation of [URL.cachesDirectory](https://github.com/apple/swift-foundation/blob/046bb36763c420545f8a14f77bb1325954394f8d/Sources/FoundationEssentials/URL/URL.swift#L2491).
- Adds a test to ensure `existingDebugLogs()` filters out non existing urls. I don't like this test very much but it is hard to improve without more significant changes.

### Testing

0. Clean install app, don't have Mail.app configured, have Gmail installed.
1. Login.
2. Go to Account > Settings > Advanced > Debug Report.
3. Tap "Send report to Wire".
4. Tap "OK" on alert informing you that no mail clients are set up.
5. Select Gmail.
6. Gmail compose email sheet appears with bug reports attached.

### Checklist
